internal/cluster/distribution/eks/service.go

```diff
@@ -85,15 +84,76 @@ type NodePoolUpdateDrainOptions struct {
 
 // NodePool encapsulates information about a cluster node pool.
 type NodePool struct {
-	Name         string            `mapstructure:"name"`
-	Labels       map[string]string `mapstructure:"labels"`
-	Size         int               `mapstructure:"size"`
-	Autoscaling  Autoscaling       `mapstructure:"autoscaling"`
-	VolumeSize   int               `mapstructure:"volumeSize"`
-	InstanceType string            `mapstructure:"instanceType"`
-	Image        string            `mapstructure:"image"`
-	SpotPrice    string            `mapstructure:"spotPrice"`
-	SubnetID     string            `mapstructure:"subnetId"`
+	Name          string            `mapstructure:"name"`
+	Labels        map[string]string `mapstructure:"labels"`
+	Size          int               `mapstructure:"size"`
+	Autoscaling   Autoscaling       `mapstructure:"autoscaling"`
+	VolumeSize    int               `mapstructure:"volumeSize"`
+	InstanceType  string            `mapstructure:"instanceType"`
+	Image         string            `mapstructure:"image"`
+	SpotPrice     string            `mapstructure:"spotPrice"`
+	SubnetID      string            `mapstructure:"subnetId"`
+	Status        NodePoolStatus    `mapstructure:"status"`
+	StatusMessage string            `mapstructure:"statusMessage"`
+}
+
+// NodePoolStatus represents the possible states of a node pool.
+type NodePoolStatus string
+
+const (
+	// NodePoolStatusCreating is the status used when the node pool resources
+	// are being provisioned.
+	NodePoolStatusCreating NodePoolStatus = "CREATING"
+
+	// NodePoolStatusDeleting is the status used when the node pool resources
+	// are being removed.
+	NodePoolStatusDeleting NodePoolStatus = "DELETING"
+
+	// NodePoolStatusEmpty is the status used when the node pool status needs to
+	// be explicitly set to an empty value. This is also the type's default
+	// value.
+	NodePoolStatusEmpty NodePoolStatus = ""
+
+	// NodePoolStatusCreating is the status returned when the node pool
+	// is in an invalid state or an operation cannot be performed on it.
+	NodePoolStatusError NodePoolStatus = "ERROR"
+
+	// NodePoolStatusCreating is the status returned when the node pool
+	// is in a healthy, idle state.
+	NodePoolStatusReady NodePoolStatus = "READY"
+
+	// NodePoolStatusUnknown is the status returned when the node pool cannot be
+	// examined.
+	NodePoolStatusUnknown NodePoolStatus = "UNKNOWN"
+
+	// NodePoolStatusUpdating is the status returned when the node pool
+	// resources are being changed.
+	NodePoolStatusUpdating NodePoolStatus = "UPDATING"
+)
+
+// NewNodePoolStatusFromCFStackStatus translates a CloudFormation stack status
+// into a node pool status.
+func NewNodePoolStatusFromCFStackStatus(cfStackStatus string) (nodePoolStatus NodePoolStatus) {
+	switch {
+	case strings.HasSuffix(cfStackStatus, "_COMPLETE"):
+		if cfStackStatus == "DELETE_COMPLETE" { // Note: CF stack is deleted, but DB entry is still existing.
```

Comment:
There are some nice constants for these values like: `ChangeSetStatusDeleteComplete` in the `cloudformation` package of the AWS SDK. I guess because the substring matching they are not used, just a notic.

Reply:
That's correct, also I was unsure of whether we want to references those from here, but I'm going to replace the exact values with those constants.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/e5f64f3bea2496abc75b55cb13b0cee1651d9793..f92f6f4a1158cc11e4b9a495b58a3bf3cb952702).